### PR TITLE
B 31 edit congratulation by

### PR DESCRIPTION
**Pull request by AlexKievDeveloper**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -170,8 +170,8 @@
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-test</artifactId>
-            <version>2.3.4.RELEASE</version>
+            <artifactId>spring-boot-test-autoconfigure</artifactId>
+            <version>2.3.5.RELEASE</version>
             <scope>test</scope>
```
Comment (Yuliia-Sh) [inline pom.xml:175]: why autoconfigure is  needed?

File: pom.xml [hunk 2]
```diff
@@ -170,8 +170,8 @@
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-test</artifactId>
-            <version>2.3.4.RELEASE</version>
+            <artifactId>spring-boot-test-autoconfigure</artifactId>
+            <version>2.3.5.RELEASE</version>
```
Comment (RomanKlishch) [inline pom.xml:174]: это лучше вынести в проперти

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java [hunk 3]
```diff
@@ -132,7 +121,71 @@ public void deleteById(long congratulationId, long userId) {
...
         params.put("congratulation_id", congratulationId);
         namedJdbcTemplate.update(DELETE_BY_ID, params);
+    }
 
+    @Override
+    public void deleteLinksById(List<Link> linkIdToDelete, long congratulationId) {
+        deleteFilesFromLinks(linkIdToDelete, congratulationId);
+
+        jdbcTemplate.update(connection -> {
+            connection.setAutoCommit(false);
```
Comment (Yuliia-Sh) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java:131]: can be used one query - delete from ... where id in (:id1, :id2, ...)

**Review by RomanKlishch — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (AlexKievDeveloper) [reply pom.xml:175]: > why autoconfigure is needed?

It's not obligatory. I can remove this.

File: src/main/java/com/greetingcard/dao/jdbc/mapper/LinksRowMapper.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+public class LinksRowMapper implements ResultSetExtractor<List<Link>> {
+
+    @Override
+    public List<Link> extractData(ResultSet resultSet) throws SQLException, DataAccessException {
+        List<Link> linkList = new ArrayList<>();
+
+        if (!resultSet.next()) {
+            return linkList;
+        }
+
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/dao/jdbc/mapper/LinksRowMapper.java:22]: We can delete this if and use while without do

Comment (OleksandrGubariev) [reply src/main/java/com/greetingcard/dao/jdbc/mapper/LinksRowMapper.java:22]: We return list. We have the same return.

File: src/main/java/com/greetingcard/web/controller/CongratulationController.java [hunk 5]
```diff
@@ -77,7 +86,26 @@ public CongratulationController(CongratulationService congratulationService) {
         return ResponseEntity.status(HttpStatus.CREATED).build();
     }
 
-    @DeleteMapping(value = "/{id}")
+    @PutMapping("/{id}")
+    public ResponseEntity<?> editCongratulation(@RequestParam(required = false) MultipartFile[] files_image,
+                                                @RequestParam(required = false) MultipartFile[] files_audio,
+                                                @RequestParam String json,
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/web/controller/CongratulationController.java:92]: What parameters does this string contains.

File: src/main/java/com/greetingcard/dao/jdbc/mapper/LinksRowMapper.java (hunk 4 above)
Comment (AlexKievDeveloper) [reply src/main/java/com/greetingcard/dao/jdbc/mapper/LinksRowMapper.java:22]: > We can delete this if and use while without do

I changed it to while and using one return

File: src/main/java/com/greetingcard/web/controller/CongratulationController.java (hunk 5 above)
Comment (AlexKievDeveloper) [reply src/main/java/com/greetingcard/web/controller/CongratulationController.java:92]: > What parameters does this string contains.

message, youtube

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java (hunk 3 above)
Comment (AlexKievDeveloper) [reply src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java:131]: > where id in (:id1, :id2, ...)

I changed method accordinaglly to your advice. Thanks.

**Review by Yuliia-Sh — APPROVED** [review]

**Review by OleksandrGubariev — APPROVED** [review]

**Pull request merged**